src/Forms/Input/Scanner.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/*
+Gibbon, Flexible & Open School System
+Copyright (C) 2010, Ross Parker
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+namespace Gibbon\Forms\Input;
+use Gibbon\Forms\Element;
+
+/**
+ * Scanner
+ *
+ * @version v23
+ * @since   v23
+ */
+class Scanner extends Input
```

Comment:
If this class were to extend TextField rather than Input, then you don't need to duplicate the existing methods, you can just add the new ones that you need. Keeps the code DRY :)

Reply:
Ah yes, I /think/ I get it? Working with classes is not my forte.